### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by Vishruta-Patil**
Features -
1. Fetch wishlist of user after login
2. Adding items to wishlist from product list page
3. Adding items to wishlist from home page
4. Removing items from wishlist

PS - Ignore commented code, need for my reference

### CODE REVIEW

File: src/pages/auth/signin.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        placeholder="Enter your password"
+        onChange={(e) =>
+          setCredentials({ ...credentials, password: e.target.value })
+        }
+      />
+      {/* <input
+          type="password"
+          class="input-box"
+          placeholder="Confirm your password"
+        /> */}
```
Comment (tewarimohit) [inline src/pages/auth/signin.js:60]: remove unnecessary comments..

File: src/reducer/user/userReducer.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    switch (action.type) {
+        case USER_LOADING:
+            return {
+                ...state,
+                userLoading: !state.userLoading
+            }
+        case LOGIN_STATUS:
+            return {
+                ...state,
+                // loginStatus: localStorage.getItem("token") !== null
```
Comment (tewarimohit) [inline src/reducer/user/userReducer.js:13]: remove comments..

File: src/styles/home.css [hunk 3]
```diff
@@ -73,8 +73,9 @@
   opacity: 0;
 }
 
-.category-unit:hover,
-.product-unit:hover {
+.category-unit:hover
+/* .product-unit:hover */
```
Comment (tewarimohit) [inline src/styles/home.css:77]: remove commented line

File: src/styles/index.css [hunk 4]
```diff
@@ -7,4 +7,6 @@
 @import url(./login.css);
 @import url(./productList.css);
 @import url(./cart.css);
-@import url(./wishlist.css);
\ No newline at end of file
+@import url(./wishlist.css);
+
+/* https://freefrontend.com/css-loaders/ */
```
Comment (tewarimohit) [inline src/styles/index.css:12]: remove commented line

File: src/styles/wishlist.css [hunk 5]
```diff
@@ -27,9 +27,48 @@
 }
 
 .wishlist-icon {
+  color: var(--secondary-color);
+}
+
+/* .wishlist-icon:hover {
+  color: var(--primary-color);
+} */
```
Comment (tewarimohit) [inline src/styles/wishlist.css:35]: remove commented lines

**Review by tewarimohit — APPROVED** [review]
Remove commented lines... otherwise good to merge..

File: src/components/common/ProductCard/index.js [hunk 6]
```diff
@@ -1,23 +1,55 @@
...
-            <button class="hero-btn product-unit-btn">Buy Now</button>
-            <button class="outline-btn product-unit-btn">Add To Cart</button>
-            </div>
-        </div>
+const ProductCard = ({ item, clickHandler, from }) => {
+  const [activeClass, setActiveClass] = useState("wishlist-icon")
+  const {state, dispatch} = useWishList()
+
+  const btnHandler = () => {
+    // dispatch({type: WISHLIST_STATUS_CLASS})
```
Comment (prathmeshjagtap) [inline src/components/common/ProductCard/index.js:12]: you can remove commented code

File: src/components/common/ProductCard/index.js [hunk 7]
```diff
@@ -1,23 +1,55 @@
...
+          onClick={btnHandler}
+        >
+          {from ?
+          <span class="material-icons wishlist-in-icon"> favorite </span>
+          :  <span class={`material-icons ${activeClass}`}> favorite </span> //state.wishListStatusClass
 }
+        </button>      
+      </div>
+
+      {/* <p class="description-product">{item.product_desc}</p> */}
```
Comment (prathmeshjagtap) [inline src/components/common/ProductCard/index.js:41]: remove this

File: src/pages/Wishlist/index.js [hunk 8]
```diff
@@ -1,5 +1,75 @@
...
-    )
-}
\ No newline at end of file
+  const { state, dispatch } = useWishList();
+  const encodedtoken = localStorage.getItem("token");
+
+  useEffect(
+    () =>
+      (async () => {
+        {console.log("wishlist1: ", state.wishListLoader)}
```
Comment (prathmeshjagtap) [inline src/pages/Wishlist/index.js:16]: remove console log

File: src/pages/Wishlist/index.js [hunk 9]
```diff
@@ -1,5 +1,75 @@
...
+        dispatch({type: WISHLIST_LOADER})
+        {console.log("wishlist2: ", state.wishListLoader)}
```
Comment (prathmeshjagtap) [inline src/pages/Wishlist/index.js:18]: remove console.log

File: src/pages/Wishlist/index.js [hunk 10]
```diff
@@ -1,5 +1,75 @@
...
+        try {
+          const response = await axios.get("/api/user/wishlist", {
+            headers: {
+              authorization: encodedtoken,
+            },
+          });
+
+          dispatch({type: WISHLIST_LOADER})
+          {console.log("wishlist3: ", state.wishListLoader)}
```
Comment (prathmeshjagtap) [inline src/pages/Wishlist/index.js:27]: remove onsole.log

File: src/pages/Wishlist/index.js [hunk 11]
```diff
@@ -1,5 +1,75 @@
...
+  const deleteWishListHandler = async ({ item }) => {
+    try {
+      const config = {
+        headers: {
+          authorization: localStorage.getItem("token"),
+        },
+      };
+      
+      const response = await axios.delete(`/api/user/wishlist/${item._id}`, config);
+      console.log("wish res: " + response.data.wishlist)
```
Comment (prathmeshjagtap) [inline src/pages/Wishlist/index.js:46]: you can remove this

File: src/pages/home/index.js [hunk 12]
```diff
@@ -3,11 +3,41 @@ import ContentTitle from "../../components/home/contentTiltle";
...
+        headers: {
+          authorization: localStorage.getItem("token"),
+        },
+      };
+      const data = {
+        product: item,
+      };
+      await axios.post("/api/user/wishlist", data, config);
+      console.log("working from home")
+
```
Comment (prathmeshjagtap) [inline src/pages/home/index.js:25]: you can remove  this

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]
everything  else is fine

**Review by prathmeshjagtap — APPROVED** [review]

File: src/pages/home/index.js (hunk 12 above)
Comment (Vishruta-Patil) [reply src/pages/home/index.js:25]: Made all the required changes in the branch - Cart

**Pull request merged**